src/modules/NtpClient.rb

```diff
@@ -598,7 +598,9 @@ def Write
       # write settings
       return false if !go_next
 
-      @ntp_records += restrict_map_records
+      # Restrict map records are written first to not mangle the config file
+      # (bsc#983486)
+      @ntp_records = restrict_map_records + @ntp_records
```

Comment:
I don't like that it adds restrict maps to ntp_records each time Write is called, we probably could remove this line and just concatenate both arrays in records_for_write, but is preferable to do it in a separate PR.